[nix] Android builds failing due to "still alive" error

Currently Android builds randomly fail due to some form of race condition resulting in this error:
```
finding garbage collector roots...
removing stale temporary roots file '/nix/var/nix/temproots/29792'
removing stale temporary roots file '/nix/var/nix/temproots/29812'
error: cannot delete path \
  '/nix/store/0w5f4dk7zcsylq426yq8yj32y6s8lbzj-status-react-source-gradle-install' \
  since it is still alive
make: *** [Makefile:158: jsbundle-android] Error 1
```
https://ci.status.im/job/status-react/job/prs/job/android/view/change-requests/job/PR-8713/2/

I merged #8788 but it seems like the issue is still happening, though less frequently:
```
error: cannot delete path '/nix/store/qmrraqygl5dxghh35i0l9666j04ahpjm-status-react-build-jsbundle-android.drv' since it is still alive
make: *** [Makefile:159: jsbundle-android] Error 1
```
https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-8811/8/

So I just made a PR where we ignore `nix/clean.sh` failures in `nix/build.sh` when in CI: #8817

--- a/nix/build.sh
+++ b/nix/build.sh
@@ -6,8 +6,16 @@ set -e
 function cleanup() {
   # clear trapped signals
   trap - EXIT ERR INT QUIT
-  # do the actual cleanup
-  ./nix/clean.sh "${nixResultPath}"
+  # do the actual cleanup, ignore failure 
+  if ./nix/clean.sh "${nixResultPath}"; then
+    echo "Successful cleanup!"
+  elif [[ -n "${JENKINS_URL}" ]]; then
+    # in CI removing some paths can fail due to parallel builds
+    echo "Ignoring cleanup failure in CI."
+  else
+    echo "Failed cleanup!"
+    exit 1
+  fi
 }
 
 trap cleanup EXIT ERR INT QUIT

